spec/core/ics-004-channel-and-packet-semantics/README.md

```diff
@@ -561,15 +560,15 @@ function sendPacket(
     abortTransactionUnless(authenticateCapability(channelCapabilityPath(sourcePort, sourceChannel), capability))
 
     // disallow packets with a zero timeoutHeight and timeoutTimestamp
-    abortTransactionUnless(packet.timeoutHeight !== 0 || packet.timeoutTimestamp !== 0)
+    abortTransactionUnless(timeoutHeight !== 0 || timeoutTimestamp !== 0)
     
     // check that the timeout height hasn't already passed in the local client tracking the receiving chain
     latestClientHeight = provableStore.get(clientPath(connection.clientIdentifier)).latestClientHeight()
-    abortTransactionUnless(packet.timeoutHeight === 0 || latestClientHeight < packet.timeoutHeight)
+    abortTransactionUnless(timeoutHeight === 0 || latestClientHeight < timeoutHeight)
 
     // increment the send sequence counter
-    sequence = provableStore.get(nextSequenceSendPath(packet.sourcePort, packet.sourceChannel))
-    provableStore.set(nextSequenceSendPath(packet.sourcePort, packet.sourceChannel), sequence+1)
+    sequence = provableStore.get(nextSequenceSendPath(sourcePort, sourceChannel))
+    provableStore.set(nextSequenceSendPath(sourcePort, sourceChannel), sequence+1)
 
     // store commitment to the packet data & packet timeout
     provableStore.set(
```

Comment:
One thing I also noticed was that when storing the packet commitment here

```
provableStore.set(
   packetCommitmentPath(sourcePort, sourceChannel, sequence),
   hash(data, timeoutHeight, timeoutTimestamp)
)
```

We store `hash(data, timeoutHeight, timeoutTimestamp)` while in [ibc-go we store `hash(hash(data), timeoutHeight, timeoutTimestamp)`](https://github.com/cosmos/ibc-go/blob/main/modules/core/04-channel/types/packet.go#L16). Maybe we can also update this, if @AdityaSripal agrees?

Reply:
Yes this must be consistent with the ibc-go implementation